addons/building_markers/functions/fnc_set.sqf

```diff
@@ -60,12 +60,16 @@ if (_set) then {
     _marker setMarkerSizeLocal _size;
     _marker setMarkerDir getDir _object;
     _object setVariable [QGVAR(marker), _marker, true];
+    missionNamespace setVariable [_marker, _object];
+
+    [QEGVAR(common,setMarkerDrawPriority), [_marker, -1], _marker] call CBA_fnc_globalEventJIP;
 
     // Delete marker when the object is deleted
     private _eventID = _object addEventHandler ["Deleted", {
         params ["_object"];
 
-        deleteMarker (_object getVariable [QGVAR(marker), ""]);
+        private _marker = _object getVariable [QGVAR(marker), ""];
+        deleteMarker _marker;
```

Comment:
Why this change?

I understand if you would want to add a condition, such as this:
```suggestion
        private _marker = _object getVariable [QGVAR(marker), ""];
        
        if (_marker != "") then {
            deleteMarker _marker;
        };
```

Otherwise not really worth it imo.